### PR TITLE
TGP-1114: Create name page

### PR DESCRIPTION
**Pull request by yohan-oc**

### CODE REVIEW

File: app/forms/NameFormProvider.scala [hunk 1]
```diff
@@ -26,6 +26,6 @@ class NameFormProvider @Inject() extends Mappings {
   def apply(): Form[String] =
     Form(
       "value" -> text("name.error.required")
-        .verifying(maxLength(100, "name.error.length"))
+        .verifying(maxLength(70, "name.error.length"))
```
Comment (vinoocs) [inline app/forms/NameFormProvider.scala:29]: I suppose, We also need to trim the spaces -  Trim the ends and remove multiple spaces

File: app/controllers/AdviceStartController.scala [hunk 2]
```diff
@@ -42,7 +42,6 @@ class AdviceStartController @Inject() (
     Ok(view())
   }
 
-  //TODO navigate to correct page
```
Comment (besscerule) [inline app/controllers/AdviceStartController.scala]: navigation is in a separate ticket

File: app/navigation/Navigator.scala [hunk 3]
```diff
@@ -41,6 +41,7 @@ class Navigator @Inject() () {
     case CommodityCodePage      => _ => routes.HasCorrectGoodsController.onPageLoad(NormalMode)
     case HasCorrectGoodsPage    => navigateFromHasCorrectGoods
     case p: AssessmentPage      => navigateFromAssessment(p)
+    case AdviceStartPage        => _ => routes.NameController.onPageLoad(NormalMode)
```
Comment (besscerule) [inline app/navigation/Navigator.scala]: navigation is in a separate ticket

File: app/views/NameView.scala.html [hunk 4]
```diff
@@ -39,6 +39,7 @@
                 field = form("value"),
                 label = LabelViewModel(messages("name.heading")).asPageHeading()
```
Comment (besscerule) [inline app/views/NameView.scala.html]: check sizing of this - i think it's too big

File: test/controllers/AdviceStartControllerSpec.scala [hunk 5]
```diff
@@ -52,7 +52,7 @@ class AdviceStartControllerSpec extends SpecBase {
 
     val onwardRoute = Call("", "")
 
-    "must redirect to ??? when the user click continue button" in {
```
Comment (besscerule) [inline test/controllers/AdviceStartControllerSpec.scala]: this is in a separate navigation ticket

File: test/navigation/NavigatorSpec.scala [hunk 6]
```diff
@@ -288,6 +288,30 @@ class NavigatorSpec extends SpecBase {
           navigator.nextPage(AssessmentPage("id1"), NormalMode, answers) mustEqual routes.IndexController.onPageLoad
         }
       }
+
```
Comment (besscerule) [inline test/navigation/NavigatorSpec.scala:291]: This is in the navigation ticket

**Review by besscerule — CHANGES_REQUESTED** [review]

File: app/controllers/AdviceStartController.scala (hunk 2 above)
Comment (yohan-oc) [reply app/controllers/AdviceStartController.scala]: This is fixed

File: app/navigation/Navigator.scala (hunk 3 above)
Comment (yohan-oc) [reply app/navigation/Navigator.scala]: Removed :)

File: test/controllers/AdviceStartControllerSpec.scala (hunk 5 above)
Comment (yohan-oc) [reply test/controllers/AdviceStartControllerSpec.scala]: Removed :)

File: test/navigation/NavigatorSpec.scala (hunk 6 above)
Comment (yohan-oc) [reply test/navigation/NavigatorSpec.scala:291]: Updated :)

File: app/views/NameView.scala.html (hunk 4 above)
Comment (yohan-oc) [reply app/views/NameView.scala.html]: This is fixed

**Review by ChrisCookOC — APPROVED** [review]

**Review by besscerule — APPROVED** [review]

**Review by vinoocs — APPROVED** [review]

**Pull request merged**